cron_ext/extension.py

```diff
@@ -0,0 +1,232 @@
+from __future__ import annotations
+
+import json
+import os
+import re
+import stat
+import subprocess
+import sys
+from contextlib import suppress
+from hashlib import sha256
+from pathlib import Path
+from typing import Callable, Iterable
+
+import structlog
+from cached_property import cached_property
+from meltano.edk import models
+from meltano.edk.extension import ExtensionBase
+
+from cron_ext import APP_NAME
+
+log = structlog.get_logger(APP_NAME)
+
+
+def run_subprocess(
+    args: tuple[str],
+    error_message: str,
+    exit_on_nonzero_returncode: bool = True,
+) -> subprocess.CompletedProcess:
+    try:
+        proc = subprocess.run(
+            args,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        )
+    except Exception:
+        log.exception(error_message)
+        sys.exit(1)
+    if proc.returncode:
+        if exit_on_nonzero_returncode:
+            log.critical(error_message)
+            sys.exit(proc.returncode)
+        log.error(error_message)
+    return proc
+
+
+class Cron(ExtensionBase):
+    full_line_comment_pattern = re.compile(r"^\s*#.*$")
+    entry_pattern = re.compile(r"(?i)^.+?'(?P<path>(?:/.*)*/(?P<name>.*)\.sh)'.*$")
+
+    def invoke(self, *args, **kwargs):
+        """Invoke the underlying cli, that is being wrapped by this extension."""
+        raise NotImplementedError
+
+    @cached_property
+    def meltano_project_dir(self) -> Path:
+        return Path(os.environ["MELTANO_PROJECT_ROOT"]).resolve()
+
+    @cached_property
+    def cron_ext_dir(self) -> Path:
+        path = self.meltano_project_dir / ".meltano" / "cron-ext"
+        path.mkdir(parents=True, exist_ok=True)
+        return path
+
+    @cached_property
+    def meltano_schedule(self) -> dict[str, str]:
+        return json.loads(
+            run_subprocess(
+                ("meltano", "schedule", "list", "--format=json"),
+                "Unable to list Meltano schedules.",
+            ).stdout
+        )["schedules"]
+
+    @cached_property
+    def meltano_schedule_ids(self) -> set[str]:
+        return {
+            schedule["name"]
+            for schedule in self.meltano_schedule["elt"] + self.meltano_schedule["job"]
+        }
+
+    @property
+    def crontab(self) -> str:
+        return run_subprocess(
+            ("crontab", "-l"),
+            "Unable to list crontab entries.",
```

Comment:
@WillDaSilva are you still planning on adding support for:

> https://github.com/meltano/meltano/issues/3437#issuecomment-1239767010: 
> I, personally, would have preferred we not actually attempt to install/manage per user crontab's and just printed out the entries instead (because I think that might be more friendly when folks are deploying this via ansible/salt/etc), [...]
> @pandemicsyn That will be supported by running meltano invoke cron list.

... in a follow up ?

Reply:
I missed that when I reviewed that thread. I've created https://github.com/meltano/cron-ext/issues/9 to address it.